legal-api/migrations/versions/540e5427636b_add_naics_tables.py

```diff
@@ -0,0 +1,111 @@
+
+"""
+from flask import current_app
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.sql.schema import MetaData, Table
+
+from migrations.bulk_data.naics_structures_2017_v3 import naics_structures
+from migrations.bulk_data.naics_elements_2017_v3 import naics_elements
+from migrations.bulk_data.naics_structures_2017_v3_subset_testing_only import naics_structures_test_only
+from migrations.bulk_data.naics_elements_2017_v3_subset_testing_only import naics_elements_test_only
+
+# revision identifiers, used by Alembic.
+revision = '540e5427636b'
+down_revision = '94586877d4bc'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+
+    # Notes: bulk dictionary data for naics_structure and naics_element data used for the bulk inserts below were
+    # generated via a Jupyter notebook(/lear/tests/data/naics/create_alembic_import_data_from_naics_csvs.ipynb).
+    # The dictionary data are immported from migrations.bulk_data to make this migration file more readable and
+    # also easier to work with.
+    #
+    # Different data sets are used to overcome the slow performance of using the actual NAICS dataset for testing
+    # purposes.  As such when migrations are performed during test runs, only a small subset of data is used to load
+    # the test database for tests.
+
+    is_test_mode = current_app.config.get('TESTING', False)
+    print(f'is_test_mode: {is_test_mode}')
+
+    # naics_structures table
+    op.create_table('naics_structures',
+                    sa.Column('id', sa.Integer(), nullable=False),
+                    sa.Column('level', sa.Integer(), nullable=False),
+                    sa.Column('hierarchical_structure', sa.String(length=25), nullable=False),
+                    sa.Column('code', sa.String(length=10), nullable=False),
+                    sa.Column('year', sa.Integer(), nullable=False),
+                    sa.Column('class_title', sa.String(length=150), nullable=False),
+                    sa.Column('superscript', sa.String(length=5), nullable=True),
+                    sa.Column('class_definition', sa.String(length=5100), nullable=False),
+                    sa.PrimaryKeyConstraint('id')
+                    )
+    op.create_index(op.f('ix_naics_structure_level'), 'naics_structures', ['level'], unique=False)
+    op.create_index(op.f('ix_naics_structure_code'), 'naics_structures', ['code'], unique=False)
+    op.create_index(op.f('ix_naics_structure_year'), 'naics_structures', ['year'], unique=False)
+    op.create_index(op.f('ix_naics_structure_class_title'), 'naics_structures', ['class_title'], unique=False)
+    op.create_index(op.f('ix_naics_structure_class_defn'), 'naics_structures', ['class_definition'], unique=False)
+
+    meta = MetaData(bind=op.get_bind())
+    meta.reflect(only=('naics_structures',))
+    naics_structures_table = Table('naics_structures', meta)
+
+    naics_structures_to_insert = naics_structures_test_only if is_test_mode else naics_structures
```

Comment:
Note that a different set of data is loaded when unit tests are running.  Note sure if we are ok with this but I've done it this way because tests were taking a long time to run when the full set of data needed to be loaded.